packages/contracts/docs/osx/02-how-to-guides/01-dao/02-action-execution.md

```diff
@@ -1,15 +1,77 @@
+
+## How to grant the Execute Permission
+
+Usually, the `EXECUTE_PERMISSION` is granted to a governance plugin of the DAO so that only the approved proposals can be executed. For example, we'd grant the `EXECUTE_PERMISSION` to the address of the Multisig Plugin, so that when a proposal is approved by the required members of the multisig, the plugin is able to call the `execute` function on the DAO in order to get the actions executed.
+
+To grant the `EXECUTE_PERMISSION` to an address, you'll want to call on the `PermissionManager`'s [`grant` function](https://github.com/aragon/osx/blob/develop/packages/contracts/src/core/permission/PermissionManager.sol#L105) and pass it 4 parameters:
+
+- `where`: the address of the contract containing the function `who` wants access to
+- `who`: the address (EOA or contract) receiving the permission
+- `permissionId`: the permission identifier the caller needs to have in order to be able to execute the action
+- `condition`: the condition that will be asked before authorizing the call to happen
+
+:::caution
+You don't want to grant `EXECUTE_PERMISSION` to any random address, since this gives the address access to execute any action on behalf of the DAO. We recommend you only grant `EXECUTE_PERMISSION` to governance plugins to ensure the safety of your assets.
+:::
+
+## Examples
+
+### Calling a DAO Function
+
+Imagine you want to call an internal function inside the `DAO` contract, for example, to manually [grant or revoke a permission](../../01-how-it-works/01-core/02-permissions/index.md). The corresponding `Action` and `execute` function call look as follows:
+
+```solidity
+function exampleGrantCall(
+  DAO dao,
+  bytes32 _callId,
+  address _where,
+  address _who,
+  bytes32 _permissionId
+) {
+  // Create the action array
+  IDAO.Action[] memory actions = new IDAO.Action[](1);
+  actions[0] = IDAO.Action({
+    to: address(dao),
+    value: 0,
+    data: abi.encodeWithSelector(PermissionManager.grant.selector, _where, _who, _permissionId)
+  });
+
+  // Execute the action array
+  (bytes[] memory execResults, ) = dao.execute({
+    _callId: _callId,
+    _actions: actions,
+    _allowFailureMap: 0
+  });
+}
+```
+
+Here we use the selector of the [`grant` function](../../03-reference-guide/core/permission/PermissionManager.md/#external-function-grant). To revoke the permission, the selector of the [`revoke` function](../../03-reference-guide/core/permission/PermissionManager.md/#external-function-revoke) must be used.
+
+If the caller possesses the [`ROOT_PERMISSION_ID` permission](../../01-how-it-works/01-core/02-permissions/index.md#permissions-native-to-the-dao-contract) on the DAO contract, the call becomes simpler and cheaper:
+
+```solidity
+function exampleGrant(DAO dao, address _where, address _who, bytes32 _permissionId) {
+  dao.grant(_where, _who, _permissionId); // For this to work, the `msg.sender` needs the `ROOT_PERMISSION_ID`
+}
+```
+
+:::caution
```

Comment:
Maybe we should move this warning before the code snippet. I fear that people only read the  "it becomes simpler and cheaper" part and then do it but don't read the warning below 😅🙏

Reply:
Haha nice shout. Done!